2020-06-06 01:00:08,697 ERROR: Exception on /all_users [GET] [in c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\ajshe\microblog2\app\main\routes.py", line 39, in all_users
    profiles = user.order_by(user.last_seen.desc()).paginate(page, current_app.config['POSTS_PER_PAGE'], False) #gets the postfrom users followed pages, .paginate is used to retrieve only the disired page of post results
AttributeError: 'User' object has no attribute 'order_by'
2020-06-06 01:01:09,840 INFO: Microblog Startup [in C:\Users\ajshe\microblog2\app\__init__.py:74]
2020-06-06 01:01:11,633 ERROR: Exception on /all_users [GET] [in c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\ajshe\microblog2\app\main\routes.py", line 36, in all_users
    users = User.query.order_by(last_seen.desc()).all()
NameError: name 'last_seen' is not defined
2020-06-06 01:01:33,439 INFO: Microblog Startup [in C:\Users\ajshe\microblog2\app\__init__.py:74]
2020-06-06 01:01:35,204 ERROR: Exception on /all_users [GET] [in c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\ajshe\microblog2\app\main\routes.py", line 39, in all_users
    profiles = user.paginate(page, current_app.config['POSTS_PER_PAGE'], False) #gets the postfrom users followed pages, .paginate is used to retrieve only the disired page of post results
AttributeError: 'User' object has no attribute 'paginate'
2020-06-06 01:01:47,417 ERROR: Exception on /all_users [GET] [in c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\ajshe\microblog2\app\main\routes.py", line 39, in all_users
    profiles = user.paginate(page, current_app.config['POSTS_PER_PAGE'], False) #gets the postfrom users followed pages, .paginate is used to retrieve only the disired page of post results
AttributeError: 'User' object has no attribute 'paginate'
2020-06-06 01:01:55,507 ERROR: Exception on /all_users [GET] [in c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\ajshe\microblog2\app\main\routes.py", line 39, in all_users
    profiles = user.paginate(page, current_app.config['POSTS_PER_PAGE'], False) #gets the postfrom users followed pages, .paginate is used to retrieve only the disired page of post results
AttributeError: 'User' object has no attribute 'paginate'
2020-06-06 01:02:51,147 INFO: Microblog Startup [in C:\Users\ajshe\microblog2\app\__init__.py:74]
2020-06-06 01:02:52,938 ERROR: Exception on /all_users [GET] [in c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py:1891]
Traceback (most recent call last):
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\_compat.py", line 39, in reraise
    raise value
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask\app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "c:\users\ajshe\microblog2\venv\lib\site-packages\flask_login\utils.py", line 272, in decorated_view
    return func(*args, **kwargs)
  File "C:\Users\ajshe\microblog2\app\main\routes.py", line 39, in all_users
    profiles = user.paginate(page, current_app.config['POSTS_PER_PAGE'], False) #gets the postfrom users followed pages, .paginate is used to retrieve only the disired page of post results
AttributeError: 'User' object has no attribute 'paginate'
2020-06-06 01:03:53,014 INFO: Microblog Startup [in C:\Users\ajshe\microblog2\app\__init__.py:74]
